readme.md

```diff
@@ -15,834 +14,1113 @@
-        (wherever possible)
+function baz() {
+	// ...	
+}
 
-    Node:
+function quix() {
+	// ...
+}
 
-        elem.nodeType == 1
+CKEDITOR.module2 = {
+	doSomething: function() {
+		// Note that despite this function is used in another module later,
+		// it has been correctly declared as close as possible to the first usage.
+		foo();
+	}
+}
+```
 
-    null:
+Note that the same practice applies to objects:
 
-        variable === null
+```javascript
+CKEDITOR.module1 = {
+	foo: function() {
+		this.bar();
+	},
 
-    null or undefined:
+	bar: function() {
+		// ...
+	},
 
-        variable == null
+	baz: function() {
+		// ...
+	}
+};
+```
 
-    undefined:
+#### 2.2.3.3. Configuration members
 
-        Global Variables:
+Configuration members exposed by `CKEDITOR.config` object should be moved at the end of the code file. It's adapted practice in CKEditor code base which improves configuration options discoverability.
 
-            typeof variable == 'undefined'
-            // Or:
-            variable === undefined
+```javascript
+CKEDITOR.plugins.add( 'myplugin', definition );
 
-        Properties:
+// ... some code goes here ...
+// ... more lines of code ...
+// ... just don't wake up Cthulhu ...
 
-            object.prop === undefined
-            'prop' in object
-            !object.prop // If you know what you do.
+CKEDITOR.config.foo = 'foo';
+CKEDITOR.config.bar = 'bar';
+```
 
+### 2.2.4. Quotes
 
-    B. Coerced Types
+Always use single quotes in JavaScript, but double quotes in HTML.
 
-    Use `+`, `!!` and `+ ''` for type coercion whenever possible. However, your code
-    should not look like assembly or CoffeeScript so do not overuse tricks.
+## 2.3. Type checking
 
-    Here are some common cases along with coercions:
+### 2.3.1. Actual Types
 
+String:
 
-    ```javascript
+```javascript
+typeof variable === 'string'
+```
 
-    // 3.B.2.1
+Number:
 
-    var number = 1,
-        string = '1',
-        bool = false;
+```javascript
+typeof variable === 'number'
+```
 
-    number;
-    // 1
+Boolean:
 
-    number + '';
-    // '1'
+```javascript
+typeof variable === 'boolean'
+```
 
-    string;
-    // '1'
+Object:
 
-    +string;
-    // 1
+```javascript
+typeof variable === 'object'
+```
 
-    +string++;
-    // 1
+Array:
 
-    string;
-    // 2
+```javascript
+// In CKEditor:
+CKEDITOR.tools.isArray( arrayLikeObject )
+// In other cases:
+Array.isArray( arrayLikeObject )
+(wherever possible)
+```
 
-    bool;
-    // false
+Node:
 
-    +bool;
-    // 0
+```javascript
+// In CKEditor:
+elem.type == CKEDITOR.NODE_ELEMENT
+// In other cases:
+elem.nodeType == 1
+```
 
-    bool + '';
-    // 'false'
-    ```
+null:
 
+```javascript
+variable === null
+```
 
-    ```javascript
-    // 3.B.2.2
+undefined:
 
-    var number = 1,
-        string = '1',
-        bool = true;
+```javascript
+variable === undefined
+```
 
-    string === number;
-    // false
+#### 2.3.2. Coerced Types
```

Comment:
I would say it's bad practice, but yeah, pretty common. We should rather use `Boolean( x )` instead for explicitness. But let's discuss it because AFAIK I'm on losing side here (with the team).